tests/test_fetcher.py

```diff
@@ -534,3 +536,27 @@ async def mock_send(node_id, request):
             if cm is not None:
                 self.assertIn(
                     "Received unknown topic or partition error", cm.output[0])
+
+    @run_until_complete
+    async def test_solitary_abort_marker(self):
+        # An abort marker may not be preceded by any aborted messages
+
+        builder = DefaultRecordBatchBuilder(
+            magic=2, compression_type=0, is_transactional=1,
+            producer_id=3, producer_epoch=1, base_sequence=-1,
+            batch_size=999)
+        buffer = builder.build()
+
+        records = MemoryRecords(bytes(buffer))
+
+        partition_recs = PartitionRecords(
+            tp=TopicPartition('test-topic', 0),
+            records=records,
+            aborted_transactions=[],
+            fetch_offset=0,
+            key_deserializer=None,
+            value_deserializer=None,
+            check_crcs=True,
+            isolation_level=READ_COMMITTED)
+
+        self.assertEqual(len(list(partition_recs)), 0)
```

Comment:
This test doesn't fail if I switch back to `self._aborted_producers.remove()` when I run it locally. Does it work for you?

Reply:
You are right, let me try to figure out why that is...

Reply:
I updated the test in d6cfb88. The problem was that the test message was not correctly constructed.

Reply:
Hm, it seems that the workaround for creating the control batch does not work with C extension. Do you have any suggestions how to best address that?

Reply:
I took a somewhat lazy approach to address this in 35a843a. Let me know what you think is the "right" way to go about this.

Reply:
I still don't get what you're trying to do in test. Could you explain it please, better directly in comments?  
There is always a naïve approach to disable the test when it's run with C extension (use `aiokafka.util.NO_EXTENSIONS`), but I'd try to find a way that works for both cases.

Reply:
> I took a somewhat lazy approach to address this in 35a843a

It looks ok to me. But explanation in comments would be appreciated anyway.

Reply:
Added explanatory comments in e418952.